mozdownload/parser.py

```diff
@@ -56,6 +56,7 @@ def handle_data(self, data):
         if not self.active_url:
             return
 
-        name = urllib.quote(data.strip('/'))
-        if self.active_url == name:
+        name = data.strip('/')
```

Comment:
Lets get rid of stripping the final slash from here and `handle_starttag`. Instead lets do it when we call `self.entries.append`. That way we do not need the extra two variables (name, name_quoted) defined here. Lets pass this directly into the tuple.